crates/autopilot/src/infra/persistence/mod.rs

```diff
@@ -135,12 +137,207 @@ impl Persistence {
+                .await
+                .context("fetch scores")?
+                // if score is missing, no competition / auction exist for this auction_id
+                .ok_or(error::Solution::MissingScore)?;
+
+            (
+                H160(scores.winner.0).into(),
+                competition::Score::new(
+                    big_decimal_to_u256(&scores.winning_score)
+                        .ok_or(error::Solution::DbConversion("score"))?
+                        .into(),
+                )?,
+            )
+        };
+
+        // TODO: stabilize the solver competition table to get promised solution.
+        // https://github.com/cowprotocol/services/issues/2716
+        // for now, this is a `good enough` solution to unblock other tasks.
+        let solver_competition = database::solver_competition::load_by_id(&mut ex, auction_id)
+            .await
+            .context("load solver competition")?
+            .ok_or(error::Solution::SolverCompetition(anyhow::anyhow!(
+                "missing solver competition"
+            )))?;
+        let competition: model::solver_competition::SolverCompetitionDB =
+            serde_json::from_value(solver_competition.json).map_err(|_| {
+                error::Solution::SolverCompetition(anyhow::anyhow!("json conversion"))
+            })?;
+        let winning_solution =
+            competition
+                .solutions
+                .last()
+                .ok_or(error::Solution::SolverCompetition(anyhow::anyhow!(
+                    "empty solutions"
+                )))?;
+        let mut orders = HashMap::new();
+        for order in winning_solution.orders.iter() {
+            match order {
+                model::solver_competition::Order::Colocated {
+                    id,
+                    sell_amount,
+                    buy_amount,
+                } => {
+                    orders.insert(
+                        domain::OrderUid(id.0),
+                        competition::TradedAmounts {
+                            sell: (*sell_amount).into(),
+                            buy: (*buy_amount).into(),
+                        },
+                    );
+                }
+                model::solver_competition::Order::Legacy {
+                    id: _,
+                    executed_amount: _,
+                } => {
+                    return Err(error::Solution::SolverCompetition(anyhow::anyhow!(
+                        "Legacy order"
+                    )))
+                }
+            }
+        }
+        let mut prices = HashMap::new();
+        for (token, price) in winning_solution.clearing_prices.clone().into_iter() {
+            prices.insert(
+                token.into(),
+                domain::auction::Price::new(price.into())
+                    .map_err(|_| error::Solution::Price(domain::auction::InvalidPrice))?,
+            );
+        }
+        Ok(competition::Solution::new(0, winner, score, orders, prices))
```

Comment:
Using 0 as an id here is 🤡 and an indicator that we maybe shouldn't reuse the competition struct.